src/effects/native/gatereffect.cpp

```diff
@@ -0,0 +1,236 @@
+#include "effects/native/gatereffect.h"
+
+namespace {
+    // Gain when the gate is open, higher than 1 to still have a decent level
+    constexpr double kMaxGain        = 1.5; 
+    // Attack slope
+    constexpr double kBaseAttackInc  = 0.001;
+    // Release slope
+    constexpr double kBaseReleaseInc = 0.0005;
+    }
+
+// static
+QString GaterEffect::getId() {
+    return "org.mixxx.effects.gater";
+}
+
+// static
+EffectManifest GaterEffect::getManifest() {
+    EffectManifest manifest;
+    manifest.setId(getId());
+    manifest.setName(QObject::tr("Gater"));
+    manifest.setAuthor("The Mixxx Team");
+    manifest.setVersion("1.0");
+    manifest.setDescription("");
+
+    EffectManifestParameter* rate = manifest.addParameter();
+    rate->setId("rate");
+    rate->setName(QObject::tr("Rate"));
+    rate->setDescription(QObject::tr(""));
+    rate->setControlHint(EffectManifestParameter::ControlHint::KNOB_STEPPING);
```

Comment:
Hmm, should this really be stepping? With the Echo effect, it only acts as a stepping knob when the Quantize button parameter is enabled. Is there a use case for not rounding this to a specific fraction of a beat and sweeping it across a continuous range?